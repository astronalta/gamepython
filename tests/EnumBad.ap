

Manual < Enum {
    FRUIT
    PEAR
    APPLE
    MANGO = 99
}

main() Int {

    x = HELLO
    y = FRUIT
    z = Bob::MANGO

    ret 0
}
