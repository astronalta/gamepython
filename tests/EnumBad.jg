

Manual < Enum {
    FRUIT
    PEAR
    APPLE
    MANGO = 99
    x Int
    String
}

main() Int {
    w = Manual()
    x = HELLO
    y = FRUIT
    z = Bob::MANGO

    ret 0
}
