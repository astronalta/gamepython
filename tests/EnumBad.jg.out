EnumBad.jg:7:13: Explicit enum constants defined after auto enum constant(s)
EnumBad.jg:8:5: Cannot embed attribute 'x' in enum 'Manual'
EnumBad.jg:9:5: Cannot embed type 'String' in enum 'Manual'
EnumBad.jg:13:9: Enums do not have constructors
EnumBad.jg:14:9: Undefined constant 'HELLO'
EnumBad.jg:15:9: Undefined constant 'FRUIT'
EnumBad.jg:16:9: Undefined constant 'Bob::MANGO'
